homeassistant/components/matter/fan.py

```diff
@@ -0,0 +1,306 @@
+"""Matter Fan platform support."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from chip.clusters import Objects as clusters
+from matter_server.common.helpers.util import create_attribute_path_from_attribute
+
+from homeassistant.components.fan import (
+    DIRECTION_FORWARD,
+    DIRECTION_REVERSE,
+    FanEntity,
+    FanEntityDescription,
+    FanEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import MatterEntity
+from .helpers import get_matter
+from .models import MatterDiscoverySchema
+
+if TYPE_CHECKING:
+    pass
+
+
+FanControlFeature = clusters.FanControl.Bitmaps.Feature
+WindBitmap = clusters.FanControl.Bitmaps.WindBitmap
+FanModeSequenceEnum = clusters.FanControl.Enums.FanModeSequenceEnum
+
+PRESET_LOW = "low"
+PRESET_MEDIUM = "medium"
+PRESET_HIGH = "high"
+PRESET_AUTO = "auto"
+FAN_MODE_MAP = {
+    PRESET_LOW: clusters.FanControl.Enums.FanModeEnum.kLow,
+    PRESET_MEDIUM: clusters.FanControl.Enums.FanModeEnum.kMedium,
+    PRESET_HIGH: clusters.FanControl.Enums.FanModeEnum.kHigh,
+    PRESET_AUTO: clusters.FanControl.Enums.FanModeEnum.kAuto,
+}
+FAN_MODE_MAP_REVERSE = {v: k for k, v in FAN_MODE_MAP.items()}
+# special preset modes for wind feature
+PRESET_NATURAL_WIND = "natural_wind"
+PRESET_SLEEP_WIND = "sleep_wind"
```

Comment:
The whole wind feature seems a bit weird to me :sweat_smile: . 

I wonder if making them preset modes is a good idea. From what I understand, the Natural Wind mode is meant to emulate natural wind (presumably vary a bit?) but based on the current setting otherwise. It is like a "feature on top" of the regular setting.

Maybe better to just leave it out completely for now? Or make it a boolean configuration entity so it can get toggled manually? :thinking:

Reply:
I thought about it for a while and it fits HA presets - I see it reversed, I think its super strange they made this a separate feature. If this mode is enabled, the device handles the speed, just like with a preset.

I'm open to more ideas but making it a configuration entities doesn't make sense to me as its in fact an additional mode, impacting the primary functions of the device.

Maybe a separate input select entity but that decouples it entirely which is imo not so nice UX

Reply:
> I thought about it for a while and it fits HA presets - I see it reversed, I think its super strange they made this a separate feature. 

They made it separate because it is based on the speed setting, see:

> 4.4.5.2.1. SleepWind Value
The fan speed, based on current settings, SHALL gradually slow down to a final minimum speed.
For this process, the sequence, speeds and duration are MS.

"based on current settings", that said, the same is missing from natural wind, but I would expect that the natural wind feature too is based on the current setting. It says it is "manufacturer specific", so I guess technically it could be both? :thinking: 

> Maybe a separate input select entity but that decouples it entirely which is imo not so nice UX

I guess both could be enabled at the same time (in theory) so input select is also not ideal. But IMHO, it would be better so it can be selected separtely to the presets.

Reply:
Both can be enabled at the same time ? That is even more strange.
So, what you are proposing is a "multi select" input select entity then or 2 switch entities ?

That also feels like a weird hack to me and completely detached UX-wise for the user because we do not logically group the entities together for the user. But yeah, if wind mode is a special feature on top of fan modes and speeds we should make separate controls. I'll quickly discuss this with a vendor first.

Reply:
> Both can be enabled at the same time ? That is even more strange.

It seems device dependent, but it could be possible it seems :see_no_evil: But no idea what is out there, and if that would ever be used together, so... :man_shrugging:  

I think I'd go for two separate switch entities still, but I honestly don't have any strong feelings here. A separate input box is fine by me too.

Reply:
Like discussed just now in PM - Vendors will also treat the wind modes as presets - the fact that it is separate is just some inheritance from the zigbee clusters and in practice no device will support both sleep modes at the same time. If they ever will, we will have some refactoring to do but let's just start here and wait what happens in practice.

Reply:
Yeah agreed, it seems this is mostly treated as preset. Also, the fan speed can still be selected, even when natural or sleep wind mode is chosen, so it will still be possible to change the speed.

And ontop of all this: If there is a good reason, we can still refactor/change this.